src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -253,14 +253,29 @@ static Type monovm_typemap_java_to_managed (string java_type_name)
 
 		internal static Type? GetJavaToManagedType (string class_name)
 		{
-			Type? type = JNIEnvInit.RuntimeType switch {
+			lock (TypeManagerMapDictionaries.AccessLock) {
+				return GetJavaToManagedTypeCore (class_name);
+			}
+		}
+
+		private static Type? GetJavaToManagedTypeCore (string class_name)
```

Comment:
A nitpick, can you drop the `private`? Methods are private by default, the keyword is redundant :)